yoga/Yoga.cpp

```diff
@@ -106,6 +105,51 @@ static int YGDefaultLog(
 #undef YG_UNUSED
 #endif
 
+#ifdef _MSC_VER
+#define YG_ALLOC_ALIGNED ([](size_t a, size_t s) -> void* { \
+  return _aligned_malloc(s, a);                             \
+})
+#define YG_FREE_ALIGNED ([](void* m) -> void {              \
+  _aligned_free(m);                                         \
+})
+#else
+#define YG_ALLOC_ALIGNED ([](size_t a, size_t s) -> void* { \
+  return aligned_alloc(a, s);                               \
+})
+#define YG_FREE_ALIGNED ([](void* m) -> void {              \
+  free(m);                                                  \
+})
+#endif
+
+static YGAllocatorAllocateFunc gAllocatorAllocateFunc = YG_ALLOC_ALIGNED;
+static YGAllocatorFreeFunc gAllocatorFreeFunc = YG_FREE_ALIGNED;
+
+YOGA_EXPORT void YGSetAllocationCallbacks(
+    YGAllocatorAllocateFunc allocFunc,
+    YGAllocatorFreeFunc freeFunc) {
+  gAllocatorAllocateFunc = allocFunc;
+  gAllocatorFreeFunc = freeFunc;
+}
+
+YOGA_EXPORT void YGGetAllocationCallbacks(
+    YGAllocatorAllocateFunc* allocFunc,
+    YGAllocatorFreeFunc* freeFunc) {
+  if(allocFunc != nullptr) {
+    *allocFunc = gAllocatorAllocateFunc;
+  }
+  if(freeFunc != nullptr) {
+    *freeFunc = gAllocatorFreeFunc;
+  }
+}
+
+YOGA_EXPORT void* YGMemoryAllocate(size_t alignment, size_t size) {
```

Comment:
nit: Do we need to expose allocate and free as part of the public Yoga API? This already has an API for getting the allocators which are set. It doesn't seem like a good code pattern for folks to use Yoga's APIs for their own memory allocation.